### ISSUE TITLE
Tomahawk 0.8 Mac OS X Sierra does not show any album art (FLAC files on NAS)

### ISSUE THREAD

**Issue by franktopel** [tomahawk-player/tomahawk#591]
I store my music in flac format on a NAS (Zyxel NSA325 v2) and have setup an read-only user to access files from my Mac (so it doesn't create those typical Mac files in all the folders).

I do not see a single album artwork, see the attached picture. What am I to do?

![image](https://cloud.githubusercontent.com/assets/10668603/21773353/fa46457c-d68e-11e6-9b6a-de28e8c4cefc.png)

**Comment by teoanderson** [comment]
Facing the same problem here! All artwork is empty and I can't even connect to last.fm. Fortunately the song can be played. I think that is a connectivity problem causing this bug...

**Comment by PRiSTi** [comment]
I'm experience this as well. Tried to contact their support  team about it and they never responded.

**Comment by muesli** [comment]
Not being able to connect to Last.fm is the reason you're not seeing any album art. We don't read the art from local files, we fetch it from there. I'm not sure how MacOS Sierra would cause this, but I didn't experience it on previous MacOS versions. I don't currently use MacOS any longer, so someone else will have to look into this.

**Comment by felixroos** [comment]
i have the same problem with all music files on sierra. Tomahawk does not show any images at all.

**Comment by franktopel** [comment]
Why would you not use local art files? At least there should be an option to allow for local art to be used. I have all cover art directly within the music files' folder in high quality. Do I have to own and configure a Last.fm account to see album art in Tomahawk then?

**Comment by muesli** [comment]
No, their API endpoint for images doesn't require auth and so you don't need an account there or configure the Last.fm resolver.

Due to the nature of online services and resolvers we can't purely rely on local cover art (e.g. it wouldn't help if you play a song from SoundCloud), so we implemented this as a source first. Since it covers 99.9% of all "cover art needs", nobody ever bothered working on reading local cover art. If anybody's interested in taking this job, I'll happily guide them and merge it.

**Comment by franktopel** [comment]
Okay, so what are you suggesting to get this "no cover art" issue fixed? I'm a developer, but purely web frontend.

**Comment by muesli** [comment]
I'm afraid I don't have the slightest idea what would/could be causing this on MacOS Sierra only. I'd try the latest nightly from here first: http://download.tomahawk-player.org/nightly/mac/Tomahawk-latest.dmg

**Comment by mvrueden** [comment]
I had the same issue as described here. The suggestion from @muesli worked for: I can confirm that with the latest version of Tomahawk I now see album art. Thank you.

**Comment by jfml** [comment]
Nightly displays cover art of local files as well for me (0.84 doesn't) but the song notifications are not working. Should I open this as a new issue or not for nightly?

**Comment by jamiehutber** [comment]
Viewing local would still be cool I think. I have quite a few albums that are not on last-fm... Maybe if last-fm doesn't have it, check local?

**Comment by jfml** [comment]
@jamiehutber I think so, too. I also don't know if these get cached because for me they need to get reloaded everytime I start the app (it is pretty fast but you still have to wait for images to load when you're scrolling through albums).

**Comment by jamiehutber** [comment]
Ye, I've also had problems with being unable to play songs that last-fm couldn't find.

Basically this is the closest music player to winamp that I have found. I am super happy to keep making suggestions to get this to be the best on linux.

**Comment by muesli** [comment]
Covers will get cached, but only loaded once shown for the first time.

**Comment by dn5** [comment]
Just wanted to confirm here that player ain't showing album art event tho I have it on my local filesystem. This happens on MacOS Sierra, Tomahawk version 0.8.4. For some reason, the album art is not being fetched from Last.fm too.

**Comment by JDLH** [comment]
I am seeing the same problem: the cover art I so carefully added to my LAN-hosted FLAC files is not being displayed.  This is a big drawback for me.  I have lots of obscure modern classical music, so I would expect LAST.fm to be missing some of it. I would upvote the enhancement of using local cover art if present.

I am using Tomahawk player 0.8.4 on Mac OS X 10.11.

**Comment by richbos** [comment]
+1 on this. My collection has local cover art embedded in the file(s) and adding it is the first thing I do after importing a CD. I wouldn't want to have to connect to an(other) external/online service to repopulate.
